File: BookShop.Core/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace BookShop.Core.Models
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerEmail { get; set; }
        public DateTime OrderDate { get; set; }
        public string ShipToFirstName { get; set; }
        public string ShipToLastName { get; set; }
        public string ShipToStreet { get; set; }
        public string ShipToCity { get; set; }
        public string ShipToZipcode { get; set; }
        public decimal SubTotal { get; set; }
        public string Status { get; set; }
        public string PaymentIntendId { get; set; }

        public virtual ICollection<OrderItem> OrderItems { get; set; }
    }
}

File: client/src/app/data/types/order.ts
// export interface Order {
//     id: number;
//     buyerEmail: string;
//     orderDate: Date;
//     shipToFirstName: string;
//     shipToLastName: string;
//     shipToStreet: string;
//     shipToCity: string;
//     shipToZipcode: string;
//     subTotal: number;
//     status: string;
//     paymentIntendId?: any;
//     orderItems?: any;
// }


export interface OrderItem {
    id: number;
    bookId: number;
    bookName: string;
    pictureUrl: string;
    price: number;
    quantity: number;
    orderId: number;
}

export interface Order {
    id: number;
    buyerEmail: string;
    orderDate: Date;
    shipToFirstName: string;
    shipToLastName: string;
    shipToStreet: string;
    shipToCity: string;
    shipToZipcode: string;
    subTotal: number;
    status: string;
    paymentIntendId: string;
    orderItems: OrderItem[];
}